src/drivers/vmount/vmount.cpp

```diff
@@ -373,11 +375,17 @@ static int vmount_thread_main(int argc, char *argv[])
 				if (control_data_to_check != nullptr || already_active) {
 					control_data = control_data_to_check;
 					last_active = i;
+
+					if (!already_active) {
+						input_changed = true;
+					}
 				}
 			}
 
 			hrt_abstime now = hrt_absolute_time();
-			if (now - last_output_update > 10000) { // rate-limit the update of outputs
+			// Rate-limit the update of outputs, unless we just changed inputs because we wouldn't
+			// want to miss the new control data.
+			if (now - last_output_update > 10000 || input_changed) {
```

Comment:
I don't think this solves the input throttling issue completely.
I recently discovered this problem as well with ROI commands being missed because of the throttling.

In that case, `control_data` is missed even though input is not changing.

My PR to fix it is here
https://github.com/PX4/Firmware/pull/8620/files#diff-357c573ea1368bd031da4870d1fbb1feR380

Reply:
Thanks @ndepal for the note.
I think output throttling is just a bad idea. I think we should just subscribe to topics at whatever rate makes sense and then calculate it through and output it (in the same way as all other modules).
@bkueng what do you think?

Reply:
I agree. The rate limiting was introduced, because when vmount is configured to listen to mavlink commands (via `vehicle_command` message), and output mavlink (also via `vehicle_command`), the input `poll()` would wake up immediately and create a busy loop. Using `orb_set_interval` on the input side for `vehicle_command` should resolve that as well.

Reply:
@bkueng ok, who wants to change this? 😄

Reply:
I can do it, but not today anymore.